### PR TITLE
Add Profiles and refactor Addresses creation

### PR DESCRIPTION
**Pull request by jalfredr**
## Description
Creates the `Profiles` table and the linking profile_id field in `Addresses`. User creation paths have been changed to set up a default profile, and (non-empty) Address creation paths have been altered to include the default profile_id value. New model for the relevant items:
Screenshot from 2022-01-21 14-28-47

## Motivation and Context
This will allow a given user to group (or not, as they choose) any number of their Addresses under one or more Profiles. Next step will include a front-end interface to permit creation of new Profiles and the assigning/transferring of Addresses between them, as well as replacing downstream user_id values with profile_id where appropriate.

## How has this been tested?
Created new users and saw routes behave appropriately; cleared the build process.

## Have proper tags been added (for bug, enhancement, breaking change)?
- [ ] yes

## Does this PR affect any server routes?
- [ ] yes, and they are tested: [enter the % coverage here]
- [ ] yes, and they are not tested: [enter the % coverage here]
- [ ] yes, but I did not run tests
- [ ] no

### CODE REVIEW

File: server/eventHandlers/profileCreation.ts [hunk 1]
```diff
@@ -66,8 +66,15 @@ export default class extends IEventHandler {
         return;
       }
 
-      // create a new address and profile
-      addressInstance = await this._models.Address.createEmpty(chain, address);
+      // Address.createEmpty
+      const verification_token = 'NO_USER';
+      const verification_token_expires = new Date(); // expired immediately
```
Comment (zakhap) [inline server/eventHandlers/profileCreation.ts:71]: Can you explain this to me?

File: server/migrations/20211222193133-profile_refactor.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            FROM "Users" u;
+        `,
+        {
+          raw: true,
+          type: 'RAW',
+          transaction: t,
+        }
+      );
+
+      // creates a new Profile object for each User
```
Comment (zakhap) [inline server/migrations/20211222193133-profile_refactor.js:45]: change comment copy

**Review by zakhap — CHANGES_REQUESTED** [review]
Looking good! Still gotta QA a bit more

**Comment by zakhap** [comment]
Also, needs to be merged with master

File: server/models/profile.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import { ModelStatic } from './types';
+import { AddressInstance } from './address';
+
+export interface ProfileAttributes {
+  id?: number;
+  user_id: number;
+  created_at?: Date;
+  updated_at?: Date;
+  profile_name?: string;
+  email?: string;
```
Comment (zakhap) [inline server/models/profile.ts:14]: Why do we need another email?

**Comment by zakhap** [comment]
QA seems to pass for me 👀 it works on new login and connecting multiple addresses, and on moving addresses from one user to another.

File: server/eventHandlers/profileCreation.ts (hunk 1 above)
Comment (jnaviask) [reply server/eventHandlers/profileCreation.ts:71]: We literally copied the code from Addresses to the places where the code was called. Didn't change anything. Look at `createEmpty` on master.

File: server/models/profile.ts (hunk 3 above)
Comment (jnaviask) [reply server/models/profile.ts:14]: @jalfredr

Comment (jalfredr) [reply server/models/profile.ts:14]: The email in the User table is the "true" email, the single one associated with the user account for login, password retrieval, receipt of notifications, etc. The social media (including email) on the Profiles table is public-facing, permitting a user to put something (or nothing) visible to other users. I can definitely see a world where we split that off into another table downstream of Profile, but either way it should be distinct from the User version.

Comment (zakhap) [reply server/models/profile.ts:14]: okidoki. that makes sense to me, allowing people to expose a public email associated to their profile of multiple addresses

**Review by zakhap — APPROVED** [review]
Looks good to me and passes extensive QA

**Pull request merged**